app/controllers/order_items_controller.rb

```diff
@@ -0,0 +1,171 @@
+class OrderItemsController < ApplicationController
+  before_action :find_order_item, only: [:shipped, :cancelled, :update, :destroy]
+  before_action :chosen_product, only: [:create]
+
+  def create # create/update shopping cart on product pages
+    @order_item = Orderitem.find_by(order_id: order_cart, product_id: params[:product_id])
+
+    if @order_item.nil?
+      qty_limit = chosen_product.stock - params[:quantity].to_i
+
+      if qty_limit < 0 
+        flash[:error] = "You couldn't order more than the stock quantity."
+        redirect_back(fallback_location: root_path)
+        return
+      end  
+
+      @order_item = Orderitem.new(quantity: order_item_params)
+      update_orderitem 
+      
+      if @order_item.save 
+        flash[:success] = "Successfully added this item to your cart!"
+        redirect_to cart_path
+        return
+      else
+        flash[:error] = "Something went wrong, please try again!"
+        redirect_back(fallback_location: root_path)
+        return
+      end
+    else
+      @order_item.quantity += params[:quantity].to_i
+      @order_item.price += chosen_product.price * params[:quantity].to_i
+      qty_limit = chosen_product.stock - @order_item.quantity
```

Comment:
This logic is a bit "businessy". Consider how you could move it to model methods in `orderItem.rb`.